mh/pat.py

```diff
@@ -1755,15 +1792,38 @@ def sendNtcFriendAdd(self, recipient_id, accepted, seq):
         """
         if not accepted:
             return
-        data = b""
-        data += pati.lp2_string(recipient_id)
+
+        users = self.session.find_users(recipient_id, "", 1, 1)
+        if len(users) == 0:
+            self.sendAnsAlert(PatID4.NtcFriendAdd, "<LF=8><BODY><CENTER>Player is Offline.<END>", seq)
```

Comment:
This should be done before you send the [AnsFriendAccept](https://github.com/sepalani/MH3SP/blob/c47d75b4ccf8ffc3137a27068002de7d55f5e56f/mh/pat.py#L1837) packet. Because what `sendAnsAlert` is basically a cancel transaction kind of packet. So you have to fail as soon as possible. If you fail after you have finished (sent `AnsFriendAccept`) the transaction, there is no transaction to cancel. See how the packet is used in other places.

Reply:
Addressed in latest commit